fuzz/README.md

```diff
@@ -0,0 +1,79 @@
+## Fuzzing overview in holo
+
+Fuzzing is a security testing technique that involves sending random, malformed, or unexpected inputs to software to uncover vulnerabilities or bugs. It helps identify issues like crashes, memory leaks, and buffer overflows that attackers could exploit.
+
+Rust has a profile of being very secure when it comes to memory issues but that is not a complete shield from some other vectors of attack that could be used apart from memory. In Rust, two main tools exist for fuzzing: `cargo-fuzz` which is essentially a wrapper around libfuzzer and `afl.rs` which implement American Fuzzy Loop (AFL).  In holo, we use `cargo-fuzz` to do our fuzzing. 
+
+The individual fuzzes are contained inside the `fuzz/fuzz_targets` directory. We mainly fuzz the packet decoders. 
+
+Before we begin, you will need to have `cargo-fuzz` installed in your system. Therefore run:
+```
+$ cargo install cargo-fuzz
+```
+<i>We assume you already have cargo installed on your system.</i>
+
+### Listing available fuzz targets
+
+To list all the fuzz targets that we currently have, go to the holo root directory and run:
+
+```
+$ cargo fuzz list
+```
+
+Output should be something like:
+
+```
+bgp_aggregator_decode
+bgp_as_path_decode
+bgp_as_path_segment_decode
+bgp_attrs_decode
+bgp_capability_decode
+bgp_comm_decode
+bgp_ext_comm_decode
+bgp_extv6_comm_decode
+bgp_ipv4_prefix_decode
+bgp_ipv6_prefix_decode
+bgp_keepalivemsg_decode
+bgp_large_comm_decode
+bgp_message_decode
+bgp_mpreachnlri_decode
+bgp_notificationmsg_decode
+bgp_openmsg_decode
+bgp_routerefreshmsg_decode
+bgp_updatemsg_decode
+vrrp_arphdr_decode
+vrrp_etherhdr_decode
```

Comment:
Fuzzing ethernet and ARP packets isn't necessary as *holo-vrrp* only sends ARP packets, never receives them.